testSuite/scripts/test_autodetect_blob_type.py

```diff
@@ -0,0 +1,58 @@
+import filecmp
+import os
+import unittest
+import utility as util
+
+class Autodetect_Blob_Type_Scenario(unittest.TestCase):
+    # Currently, the only auto-detected blob type is page blob.
+    # Copy a VHD without specifying page blob and see what it does.
+    def test_auto_detect_blob_type_vhd(self):
+        file_name = "myVHD.vhd"
+        file_path = util.create_test_file(file_name, 4 * 1024 * 1024)
+
+        # copy VHD file without specifying page blob. Page blob is inferred for VHD, VHDX, and VMDK
+        destination_sas = util.get_object_sas(util.test_container_url, file_name)
+        result = util.Command("copy").add_arguments(file_path).add_arguments(destination_sas).add_flags("log-level",
+                                                                                                        "info"). \
+            add_flags("block-size-mb", 4).execute_azcopy_copy_command()
+        self.assertTrue(result)
+
+        # execute validator. Validator will ensure it's a page blob as validator now checks blob type before testing.
+        result = util.Command("testBlob").add_arguments(file_path).add_arguments(destination_sas). \
+            add_flags("blob-type", "PageBlob").execute_azcopy_verify()
+        self.assertTrue(result)
+
+    def test_copy_infer_blob_type_from_block_to_page_blob(self):
```

Comment:
Also, @zezha-msft , should we infer when copying from Azure Files?  Whatever we do, should be consistent.  I think the three possible ways to do that are:
(a) Always infer (even when source is Blob Storage) (this would cover all sources)  OR
(b) Always infer when source does not have a blob type (i.e. when source is not Blob Storage). This covers AWS and Azure files as sources, plus local disk., OR
(c) Infer only when uploading from local disk.

Right now we have (c) and part of (a).

Reply:
We'll automatically get (b) if we follow the approach I suggest in my overall end-of-review comment (to introduce one getTargetBlobType method)

Reply:
I think this would be a good blanket solution:

Infer blob type by default for sources that do not have a blob type, and only infer when explicitly told to from sources with a blob type.

Reply:
Followed what I suggested in the latest changes, please comment if you think a different path would be optimal or resolve if this is acceptable.

Reply:
As noted in a different comment today, I think supporting both becomes too hard to explain to users.

Reply:
Good points. Let us discuss this in the comment about the flag design.